### PR TITLE
Update JSON schema

### PR DESCRIPTION
**Pull request by goneall**
Update the JSON schema to be consistent with the current JSON example file.

- Remove the top level Document property
- Add documentNamespace

Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

### CODE REVIEW

**Comment by goneall** [comment]
Resolves issue #527

**Comment by goneall** [comment]
@SamuraiAku - this PR should resolve the issue, however, when I run it through one of the [online JSON schema validators](https://www.liquid-technologies.com/online-json-schema-validator) with the current JSON example, it seems that everything validates - even an empty JSON file.

Let me know if you have any ideas on how to improve the JSON file and I'll update the PR.

BTW - This JSON schema file is generated by the [SPDX Java tools](https://github.com/spdx/tools-java/blob/master/src/main/java/org/spdx/tools/schema/OwlToJsonSchema.java)

File: schemas/spdx-schema.json [hunk 1]
```diff
@@ -4,606 +4,95 @@
...
-                    "enum" : [ "SHA256", "SHA1", "SHA384", "MD2", "MD4", "SHA512", "MD6", "MD5", "SHA224" ]
-                  },
-                  "checksumValue" : {
-                    "description" : "The checksumValue property provides a lower case hexidecimal encoded digest value produced using a specific algorithm.",
-                    "type" : "string"
-                  }
-                },
-                "description" : "A Checksum is value that allows the contents of a file to be authenticated. Even small changes to the content of the file will change its checksum. This class allows the results of a variety of checksum and cryptographic message digest algorithms to be represented."
-              },
-              "spdxDocument" : {
```
Comment (SamuraiAku) [inline schemas/spdx-schema.json:135]: All I wanted was to change this item from spdxDocument to documentNamespace

Comment (SamuraiAku) [reply schemas/spdx-schema.json:135]: And this is buried inside the external reference property.  There's nothing for the namespace for this document

**Review by SamuraiAku — COMMENTED** [review]
Edit: Nevermind, I was mistyping when searching for the change.

~~@goneall If you want to remove the Document overall property I guess that's OK. But that's not what I was looking for. We need a field for documentNamespace, section 2.5 of the spec. There is nothing for that now.~~

**Comment by goneall** [comment]
I plan on improving the detail of the schema substantially over the next week.  I'll be posting the improvements to this PR.

**Comment by goneall** [comment]
Added licenseConcluded and licenseDeclared to schema - resolve issue #530

**Comment by goneall** [comment]
Added the missing spdxElementId from relationships to the schema - resolves issue #531

**Comment by goneall** [comment]
Schema updated with required fields which forces validation.  Resolved an issue where both `fileName` and `name` properties were on the `files` array - should only be `fileName`.

**Comment by goneall** [comment]
This schema should be good to go - I'd appreciate any reviews.

**Comment by goneall** [comment]
Renamed property describesPackages to documentDescribes - resolves issue #533

**Comment by goneall** [comment]
@SamuraiAku I would like to go ahead and merge this to resolve the following issues:
- #533 
- #531 
- #530 
- #529 
- #527 

Let me know if you agree these are resolved.  

I'll leave the other issues you identified open for further discussion.

**Comment by swinslow** [comment]
Hi @goneall, is this modifying the existing properties for JSON documents for SPDX 2.2?

I'm asking because @specter25 has been working on the JSON parsing / saving for the Golang tools for SPDX 2.2 as his GSoC project this summer, which has now concluded. Though I believe he has mostly been working from the example JSON file as the definition of SPDX JSON, so if this is just aligning the schema definition to fit with that example file, then hopefully that should be fine.

**Comment by goneall** [comment]
> is this modifying the existing properties for JSON documents for SPDX 2.2?

@swinslow The intent is not to modify any of the existing properties.  The intent is to make the current schema consistent with the JSON example file - which is our only other documentation on the JSON serialization format.

I did test [this schema](https://raw.githubusercontent.com/spdx/spdx-spec/updatejsonschema/schemas/spdx-schema.json) against the [current JSON example](https://raw.githubusercontent.com/spdx/spdx-spec/development/v2.2.1/examples/SPDXJSONExample-v2.2.spdx.json) using the [JSON Schema Lint online tool](https://jsonschemalint.com/#!/version/draft-07/markup/json) and it passed.

If you have a chance, it would be helpful to run the output of the golang tools against the same schema.  If you find any issues, we can discuss here and decide if it is a schema or tool issue.

**Comment by davaya** [comment]
In order to prevent the schema from validating everything including empty files, it's a good practice to include `"additionalProperties": false` on every object type.  I also include string size limits (I picked 5,000 characters out of the air) for extracted texts - the precise number isn't that important, but an SPDX file with megabytes or gigabytes in one extracted license text is probably not valid.  

We should have some invalid SPDX test files to ensure that the schema doesn't pass them - I'm sure there are plenty of examples from the NTIA plugfests, but haven't systematically gone through them yet.  Checking for false positives is as important as eliminating false negatives.

A [JSON schema](https://gist.github.com/davaya/ef0177d96874a6cd78ac8d41bb5578f7) generated from my [SPDX-v2.2 information model](https://gist.github.com/davaya/52e1910282f4b4fc64a286641b7b429a) also validates the current JSON example using JSON Schema Lint.

**Comment by goneall** [comment]
Thanks @davaya for the review!  I added the additionalProperties as suggested.  I didn't add the length restrictions since they are not currently included in the spec (something we should probably add).

BTW - it looks like your schema is much more complete.  We should consider using your schema in future releases.

**Comment by SamuraiAku** [comment]
> @SamuraiAku I would like to go ahead and merge this to resolve the following issues:
> 
> * [JSON schema/example has describesPackages/documentDescribes property that has no equivalent in the spec #533](https://github.com/spdx/spdx-spec/issues/533)
> * [JSON schema is missing the spdxElementId element from relationships #531](https://github.com/spdx/spdx-spec/issues/531)
> * [JSON schema is missing licenseConcluded and licenseDeclared properties in the package level #530](https://github.com/spdx/spdx-spec/issues/530)
> * [JSON schema is missing SPDXID property at the Document level #529](https://github.com/spdx/spdx-spec/issues/529)
> * [Conversion of tag-value to JSON creates a field not in the schema #527](https://github.com/spdx/spdx-spec/issues/527)
> 
> Let me know if you agree these are resolved.
> 
> I'll leave the other issues you identified open for further discussion.

@goneall Sorry for not replying earlier.  I agree that you have resolved these issues.

**Pull request merged**